lib/base/shared-object.hpp

```diff
@@ -31,23 +31,10 @@ class SharedObject
 	{
 	}
 
-	inline SharedObject(const SharedObject&) : SharedObject()
-	{
-	}
-
-	inline SharedObject(SharedObject&&) : SharedObject()
-	{
-	}
-
-	inline SharedObject& operator=(const SharedObject&)
-	{
-		return *this;
-	}
-
-	inline SharedObject& operator=(SharedObject&&)
-	{
-		return *this;
-	}
+	SharedObject(const SharedObject&) = delete;
+	SharedObject(SharedObject&&) = delete;
+	SharedObject& operator=(const SharedObject&) = delete;
+	SharedObject& operator=(SharedObject&&) = delete;
```

Comment:
Wanted to do this for the new StoppableWaitGroup requested in https://github.com/Icinga/icinga2/pull/10397#issuecomment-2898080601, but thought: